src/Functional/LessThan.php

```diff
@@ -14,7 +14,9 @@
  * Returns true if $a is strictly less than $b.
  *
  * @param mixed $b
```

Comment:
maybe

```suggestion
 * @param scalar $b
```

Reply:
I'm not sure if anyone uses this function like this but if someone compares `\DateTime`, then expecting `scalar` would result in error, which is probably a false-positive.

```php
$date = new \DateTimeImmutable('2020-01-11');
$comparator = less_than($date);

return $comparator(new \DateTimeImmutable('2050-01-01'));
```

Reply:
Good point. We could also use `scalar|DateTimeInterface`

Reply:
IDK if we should narrow down the use cases for this function. One could want to compare strings also.
I think it is up to @lstrojny to decide if the function should be more specific or not

Reply:
I don't think limiting so scalar is a good idea. DateTime is one example, others are BCMath. PHP extensions can implement overloading so there are potentially an infinite number of classes out there that are comparable.